src/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/DirectoryObjectSecurity.cs

```diff
@@ -0,0 +1,815 @@
+                        _securityDescriptor.DiscretionaryAcl.RemoveAccessSpecific(AccessControlType.Allow, sid, rule);
+                        break;
+
+                    default:
+                        throw new ArgumentOutOfRangeException(
+                            "modification",
+                            SR.ArgumentOutOfRange_Enum);
+                }
+            }
+            else if (rule.AccessControlType == AccessControlType.Deny)
+            {
+                switch (modification)
+                {
+                    case AccessControlModification.Add:
+                        //_securityDescriptor.DiscretionaryAcl.AddAccess(AccessControlType.Deny, sid, rule.AccessMask, rule.InheritanceFlags, rule.PropagationFlags, rule.ObjectFlags, rule.ObjectType, rule.InheritedObjectType);
+                        _securityDescriptor.DiscretionaryAcl.AddAccess(AccessControlType.Deny, sid, rule);
+                        break;
+
+                    case AccessControlModification.Set:
+                        //_securityDescriptor.DiscretionaryAcl.SetAccess(AccessControlType.Deny, sid, rule.AccessMask, rule.InheritanceFlags, rule.PropagationFlags, rule.ObjectFlags, rule.ObjectType, rule.InheritedObjectType);
+                        _securityDescriptor.DiscretionaryAcl.SetAccess(AccessControlType.Deny, sid, rule);
+                        break;
+
+                    case AccessControlModification.Reset:
+                        _securityDescriptor.DiscretionaryAcl.RemoveAccess(AccessControlType.Allow, sid, -1, InheritanceFlags.ContainerInherit, 0, ObjectAceFlags.None, Guid.Empty, Guid.Empty);
+                        //_securityDescriptor.DiscretionaryAcl.SetAccess(AccessControlType.Deny, sid, rule.AccessMask, rule.InheritanceFlags, rule.PropagationFlags, rule.ObjectFlags, rule.ObjectType, rule.InheritedObjectType);
+                        _securityDescriptor.DiscretionaryAcl.SetAccess(AccessControlType.Deny, sid, rule);
+                        break;
+
+                    case AccessControlModification.Remove:
+                        //result = _securityDescriptor.DiscretionaryAcl.RemoveAccess(AccessControlType.Deny, sid, rule.AccessMask, rule.InheritanceFlags, rule.PropagationFlags, rule.ObjectFlags, rule.ObjectType, rule.InheritedObjectType);
+                        result = _securityDescriptor.DiscretionaryAcl.RemoveAccess(AccessControlType.Deny, sid, rule);
+                        break;
+
+                    case AccessControlModification.RemoveAll:
+                        result = _securityDescriptor.DiscretionaryAcl.RemoveAccess(AccessControlType.Deny, sid, -1, InheritanceFlags.ContainerInherit, 0, ObjectAceFlags.None, Guid.Empty, Guid.Empty);
+                        if (result == false)
+                        {
+                            Contract.Assert(false, "Invalid operation");
```

Comment:
For this and other uses, we've standardized on Debug.Assert instead of Contract.Assert across corefx.